lib/ddtrace/contrib/rack/middlewares.rb

```diff
@@ -0,0 +1,86 @@
+require 'ddtrace/ext/app_types'
+require 'ddtrace/ext/http'
+
+module Datadog
+  module Contrib
+    # Rack module includes middlewares that are required to trace any framework
+    # and application built on top of Rack.
+    module Rack
+      # TraceMiddleware ensures that the Rack Request is properly traced
+      # from the beginning to the end. The middleware adds the request span
+      # in the Rack environment so that it can be retrieved by the underlying
+      # application. If request tags are not set by the app, they will be set using
+      # information available at the Rack level.
+      class TraceMiddleware
+        DEFAULT_CONFIG = {
+          tracer: Datadog.tracer,
+          default_service: 'rack'
+        }.freeze
+
+        def initialize(app, options = {})
+          # access tracer configurations
+          user_settings = DEFAULT_CONFIG.merge(options)
+          @app = app
+          @tracer = user_settings.fetch(:tracer)
+          @service = user_settings.fetch(:default_service)
+
+          # configure the Rack service
+          @tracer.set_service_info(
+            @service,
+            'rack',
+            Datadog::Ext::AppTypes::WEB
+          )
+        end
+
+        def call(env)
+          # get the current Rack request
+          request = ::Rack::Request.new(env)
+
+          # start a new request span and attach it to the current Rack environment;
+          # we must ensure that the span `resource` is set later
+          request_span = @tracer.trace(
+            'rack.request',
+            service: @service,
+            resource: nil,
+            span_type: Datadog::Ext::HTTP::TYPE
+          )
+          request.env[:datadog_rack_request_span] = request_span
+
+          # call the rest of the stack
+          status, headers, response = @app.call(env)
+        rescue StandardError => e
```

Comment:
It might have been better to use `Exception => e` since it's higher up in the hierarchy than `StandardError`, to make sure this catches as much as possible.